File: EJERCICIOS/python/4.py
#Escribir un algoritmo que calcule el area y volumen de un cilindro donde se 
#ingrese el radio y la altura
#Area = (2*(PI*r^2))+ ((2*PI*r)*h)
#volumen = (PI*r^2)*h
import math

def area(r,h):
    return (2*(math.pi*pow(r,2)))+((2*math.pi*r)*h)

def volumen(r,h):
    return (math.pi*pow(r,2))*h

def ingresarDatos():
    print("Ingrese el radio del Cilindro")
    r = float(input())
    print("Ingrese la altura del Cilindro")
    h = float(input())
    return (r,h)

def mostrar():
    (r,h) = ingresarDatos()
    print(f"Este el area del cilindro {area(r,h)}")
    print(f"Este el volumen del cilindro {volumen(r,h)}")

mostrar()

File: EJERCICIOS/typescript/4.ts
import readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

const rl = readline.createInterface({ input, output });

function area(radio: number, altura: number): number {
  return 2 * Math.PI * Math.pow(radio, 2) + 2 * Math.PI * radio * altura;
}

function volumen(radio: number, altura: number): number {
  return Math.PI * Math.pow(radio, 2) * altura;
}

async function ingresarDatos(): Promise<{ radio: number; altura: number }> {
  console.log('Ingrese el radio del Cilindro');
  const radio = Number(await rl.question(''));

  console.log('Ingrese la altura del Cilindro');
  const altura = Number(await rl.question(''));

  return { radio, altura };
}

async function mostrar(): Promise<void> {
  const { radio, altura } = await ingresarDatos();

  console.log(`Este el area del cilindro ${area(radio, altura)}`);
  console.log(`Este el volumen del cilindro ${volumen(radio, altura)}`);
}

await mostrar();
rl.close();

